numpy/core/src/umath/loops.c.src

```diff
@@ -2857,10 +2857,16 @@ NPY_NO_EXPORT void
         const @ftype@ in1r = ((@ftype@ *)ip1)[0];
         const @ftype@ in1i = ((@ftype@ *)ip1)[1];
         if (npy_fabs@c@(in1i) <= npy_fabs@c@(in1r)) {
-            const @ftype@ r = in1i/in1r;
-            const @ftype@ d = in1r + in1i*r;
-            ((@ftype@ *)op1)[0] = 1/d;
-            ((@ftype@ *)op1)[1] = -r/d;
+            if (in1r == 0 && in1i == 0) {
```

Comment:
I chose to mimic what `np.divide` is doing here:
https://github.com/numpy/numpy/blob/f4a4ddd56e1c3107a8b376d2e72cd5334676627a/numpy/core/src/umath/loops.c.src#L2698-L2702